### PR TITLE
Fix sidebar a11y issues

### LINKED ISSUE #1185: [Deque Analysis] Admin/Action Sidebar's "serious" accessibility issues
> Serious: This issue results in serious barriers for people with disabilities, and will partially prevent them from accessing fundamental features or content. People relying on assistive technologies will experience significant frustration as a result. Issues falling under this category are major problems, and remediation should be a priority. 

## Deque Analysis Summary

Our Admin/Action Sidebar as a total of 20 serious accessibility issues (not including color contrast which are handled in other tickets).

1. (IDs 469240, 469241, 469242, 469243, 469244, 469245, 469246, 469247, 469248, 469249, 469250, 469251, 469252) "`<ul>` and `<ol>` must only directly contain `<li>`, `<script>` or `<template>` elements" and "`<li>` elements must be contained in a `<ul>` or `<ol>`"  (Occurs on entire Admin sidebar menu, as it's made up of `<ul>` and `<li>` elements)
    * It appears our `<ul>` elements tend to contain a custom Angular template (e.g. `ds-expandable-admin-sidebar-section`) which then contains the `<li>`. We may need to find a way to stop Angular from printing out the template tag, if that's possible. Some possible hints https://stackoverflow.com/q/38716105/3750035
    * _NOTE: This same issue occurs in the menus of the Header_, see #1174, so we'd want to share a similar solution between the Header and the Admin Sidebar.
2. (ID 470071) "Expand/Collapse state of the elements is missing or incorrect." on every section header in Admin menu (e.g. "New", "Edit", "Impot", "Export", etc.)
    * Currently these are all links (`<a>`), but Deque recommends using the `aria-expanded` attribute (set to true/false) on a toggle element (like a `<button>`).  See https://www.w3.org/TR/wai-aria-1.1/#aria-expanded
3. (ID 470075, 470079, 470095, 470098, 470099) "Link disabled state is not conveyed."  on all disabled menu options in sidebar.
    * NOTE: Nothing to do here...**this will be fixed** in #1145 (by removing these disabled menu options)
4. (ID 470106) "Hidden or empty element receives focus." on all collapsed menu options.  When a section is collapsed, it's entries still get keyboard focus (when using the tab key).
     * HOW TO FIX: Fix this issue by using ONE of the following techniques:
         * If the element should not be visible or focusable, remove the element from the DOM or hide it from all users with CSS such as display:none.
         * If the element is meant to be visible but not interactive, make it visible and remove the focus state. 
         * If the element is meant to be visible and interactive, then ensure that it is both focusable and visible to all users.

Full list of issues is viewable at (requires login): https://axeauditor.dequecloud.com/test-run/0856438a-a19a-11eb-bc31-b7d5be387c86/issues?activeTab=dt-issue&page=0&pageSize=50&sortField=ordinal&sortDir=asc&filter%5Bseverity%5D=3&filter%5Btype%5D=issue&filter%5Bpage_number%5D=9&row=9

### PR DESCRIPTION
**Pull request by ybnd**
## References

* Fixes #1185

## Description

Address accessibility issues with the admin sidebar

#### List of changes in this PR

* Made HTML valid: only `<li>` elements in sidebar's `<ul>`
* Combined menu item icons & text into a single `<a>` element
* Added ARIA attributes
  * `aria-expanded`
  * `aria-disabled`
* Improved keyboard UX
  * Expand the sidebar once you "tab into it"
  * Adjusted item spacing so focus outlines are more clear
  * Double checked that items in collapsed submenus can't receive focus

#### Reviewing

* Run deque a11y check

* The sidebar should look & act the same as on `main`

  * (Except keyboard navigation, which should be improved as described above)

  

## Checklist

_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [TSLint](https://palantir.github.io/tslint/) validation using `yarn run lint`
- [x] My PR doesn't introduce circular dependencies
- [x] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new, third-party dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.

### CODE REVIEW

**Review by tdonohue — CHANGES_REQUESTED** [review]
@ybnd : Thanks for this PR!  Overall, it's much improved behavior, and running an accessibility scan (using Axe DevTools in my browser) shows the main issues are fixed.  

However, I noticed a few small issues in testing this PR
1. First, I no longer can open admin menu sections using the keyboard only.  This works fine on the demo7.dspace.org site.  Here's how to reproduce:
    * Login
    * Pin the sidebar (just to make it easier to see)
    * Using the "Tab" key on your keyboard, highlight the "New" section header.
    * Click "Enter" on keyboard.  Nothing happens (on demo site, this section will open)
    * NOTE however that if you click on a non-section link (like "Admin Search"), clicking "Enter" works correctly and brings you to the page.
2. A tiny issue. If you open one of the sections, like "New" and hover over a sub-link (e.g. "Community" or "Collection"), the title text all says "New menu section".  Ideally, all links should have their own `title`, so that they don't inherit the title of the section.   This is more an annoyance though, so if it's hard to fix, we can leave the current behavior as-is.

Beyond that it's looking good.  Thanks again!

**Comment by ybnd** [comment]
@tdonohue thanks, I'll take a look at both issues soon!

**Review by ybnd — COMMENTED** [review]
Not sure how useful it is to have it be the same as the title of the link though. 
Maybe it would make sense to add an optional `tooltip` field to the `MenuItemModel`s and use that if it exists, fall back to `text` otherwise?

File: src/app/shared/menu/menu-item/link-menu-item.component.html [hunk 1]
```diff
@@ -3,6 +3,8 @@
    [attr.aria-disabled]="!hasLink"
    [title]="item.text"
    [routerLink]="getRouterLink()"
+   (click)="$event.stopPropagation()"
```
Comment (ybnd) [inline src/app/shared/menu/menu-item/link-menu-item.component.html:6]: Due to the `[routerLink]` above `keyup.enter` events could still propagate to parent expandable sections and toggle them. 
This fixes that issue (@artlowel 👍)

**Review by tdonohue — COMMENTED** [review]
@ybnd : I gave this another test today.  The keyboard issue is fixed with the sidebar!

I did notice though that the other issue still isn't a perfect fix. 
1. Open the New menu
2. Hover over "Community" with your mouse, and you'll see the title is now `menu.section.new_community`.
3. Hover over "Collection" with your mouse, and you'll see the title is now `menu.section.new_collection`
4. This is now the case for all sub-menu items. They all have missing titles.

The strange thing is that these keys all seem to exist in our `src/assets/i18n/en.json5`.  So, I suspect you are just missing a lookup of the key value?

Beyond that, this is ready to go.  Hopefully that's a quick (possibly even one line) fix.

**Comment by ybnd** [comment]
@tdonohue Oh right, looks like I missed the `| translate`, thanks for pointing this out :+1:

**Review by tdonohue — APPROVED** [review]
👍 Looks great now, @ybnd !  Thanks for the small updates.

**Pull request merged**